shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profilePreferences/presentation/editPersonalInfo/components/EditProfileScreen.kt

```diff
@@ -0,0 +1,143 @@
+import com.vml.tutorial.plantshop.ui.theme.Typography
+
+@Composable
+fun EditProfileScreen(state: EditProfileScreenState, onEvent: (EditProfileEvent) -> Unit) {
+    if (state.showDatePickerDialog) {
+        BirthdayDialog(onDateConfirm = { onEvent(EditProfileEvent.DateConfirmed(it)) },
+            onDismissRequest = { onEvent(EditProfileEvent.DismissBirthdayDialog) })
+    }
+
+    LazyColumn {
+        item {
+            IconButton(modifier = Modifier.padding(8.dp).size(50.dp), onClick = {
+                onEvent(EditProfileEvent.NavigateBack)
+            }) {
+                Icon(
+                    imageVector = Icons.Rounded.ArrowBack,
+                    contentDescription = null,
+                    tint = MaterialTheme.colorScheme.onPrimaryContainer
+                )
+            }
+        }
+        item {
+            Column(
+                modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp),
+                verticalArrangement = Arrangement.spacedBy(16.dp)
+            ) {
+
+                ScreenTitle(
+                    UiText.StringRes(MR.strings.edit_profile_title_text).asString(),
+                    UiText.StringRes(MR.strings.edit_profile_subtitle_text).asString(),
+                    Modifier.padding(horizontal = 8.dp)
+                )
+
+                Spacer(modifier = Modifier.height(16.dp))
+
+                UserInput(
+                    state = state.firstName,
+                    placeholderText = UiText.StringRes(MR.strings.edit_profile_name_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Text, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(EditProfileEvent.FirstNameChanged(it))
+                }
+
+                UserInput(
+                    state = state.lastName,
+                    placeholderText = UiText.StringRes(MR.strings.edit_profile_lastName_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Text, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(EditProfileEvent.LastNameChanged(it))
+                }
+
+                UserInput(
+                    state = state.phoneNumber,
+                    placeholderText = UiText.StringRes(MR.strings.edit_profile_phone_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Phone, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(EditProfileEvent.PhoneNumberChanged(it))
+                }
+
+                Text(text = state.birthDate
```

Comment:
For me this field is empty, I didn't know what is it until I clicked on that :) So I think you have to check the empty value as well. e.g: state.birthDate.takeUnless { it.isNullOrEmpty() }?: placeholder text,

Reply:
Added